torch/csrc/autograd/engine.cpp

```diff
@@ -579,10 +579,9 @@ void GraphTask::exec_post_processing() {
     c10::MultiStreamGuard g(caller_current_streams_filtered);
     // WARNING: Don't use a range-for loop here because more callbacks may be
     // added in between callback calls, so iterators may become invalidated.
-    // NOLINTNEXTLINE(modernize-loop-convert)
-    for (size_t i = 0; i < final_callbacks_.size(); ++i) {
+    for (const auto& final_callback : final_callbacks_) {
       cb_lock.unlock();
-      final_callbacks_[i]();
+      final_callback();
       cb_lock.lock();
     }
```

Comment:
See comment above, range loops could not be used here because callbacks can get mutated, which I guess causes CI failures